### PR TITLE
Stats: Add image to upgrade notice

### PR DESCRIPTION
**Pull request by a8ck3n**
Part of #28729.

## Proposed changes:

Adds an image + gradient to the upgrade nudge.

SCR-20230306-keh

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

- Make sure Odyssey Stats are disabled.
- Visit Jetpack → Stats.
- Confirm the image is present in the upgrade nudge.
- Confirm the image is removed at smaller screen sizes.

**Looking for feedback!**

I'm not super happy with the gap below the image when resizing the screen but I'm unsure of how to remove it and have the image size itself properly. I've tried setting both images via CSS but then neither are drawn. I need something in the HTML to trigger proper layout.

**Updated:(per feedback)**

- Images are now handled strictly via CSS.
- Font sizes and padding adjusted.
- Missing translation calls have been added back.
- Updated image in PR description to match current status of the notice.

### CODE REVIEW

File: projects/plugins/jetpack/modules/stats.php [hunk 1]
```diff
@@ -833,18 +853,21 @@ function stats_print_odyssey_nudge( $html ) {
...
-	<div id="stats-odyssey-nudge-main" class="stats-odyssey-nudge dops-card">
-		<div class="stats-odyssey-nudge--content">
-			<div class="stats-odyssey-nudge--title"><?php esc_html_e( 'Explore the new Jetpack Stats', 'jetpack' ); ?></div>
-			<?php esc_html_e( "We've added new stats and insights in a more modern and mobile friendly experience to help you grow your site.", 'jetpack' ); ?>
-			<div class="stats-odyssey-nudge--action-bar">
-				<button class="dops-button stats-odyssey-nudge--primary-button">
+	<div id="stats-odyssey-nudge-main" class="stats-odyssey-notice">
+		<div class="stats-odyssey-notice--content">
+			<h2 class="stats-odyssey-notice--content-header">Explore the new Jetpack Stats</h2>
+			<p class="stats-odyssey-notice--content-text">We've added new stats and insights in a more modern and mobile friendly experience to help you grow your site.</p>
```
Comment (jsnmoon) [inline projects/plugins/jetpack/modules/stats.php]: What's the reason for removing the `esc_html_e` translation calls for these two lines?

File: projects/plugins/jetpack/modules/stats.php [hunk 2]
```diff
@@ -763,48 +763,67 @@ function stats_print_odyssey_nudge( $html ) {
...
 	<style>
-		.stats-odyssey-nudge {
+		.stats-odyssey-notice {
 			display: flex;
-			align-items: flex-start;
 			font-size: var( --font-body );
 
 			border: 1px solid var( --jp-gray-5 );
 			border-radius: 4px;
 			border-color: var( --jp-black );
```
Comment (jsnmoon) [inline projects/plugins/jetpack/modules/stats.php]:
```suggestion
			border-color: var( --jp-gray-5 );
			border-left-color: var( --jp-black );
```

File: projects/plugins/jetpack/modules/stats.php [hunk 3]
```diff
@@ -763,48 +763,67 @@ function stats_print_odyssey_nudge( $html ) {
...
-			padding: 24px 31px 27px 18px;
 
 			margin-top: 24px;
+			background: white;
+			position: relative;
 		}
-		.stats-odyssey-nudge--content {
-			flex-grow: 1;
+		.stats-odyssey-notice--content {
+			padding: 24px 0 27px 18px;
```
Comment (jsnmoon) [inline projects/plugins/jetpack/modules/stats.php]: image

```suggestion
			padding: 24px 0 24px 30px;
```

File: projects/plugins/jetpack/modules/stats.php [hunk 4]
```diff
@@ -763,48 +763,67 @@ function stats_print_odyssey_nudge( $html ) {
...
+			font-size: 2em;
 		}
-		.stats-odyssey-nudge--title {
-			font-weight: 600;
+		.stats-odyssey-notice--content-header {
+			font-size: 16px;
```
Comment (jsnmoon) [inline projects/plugins/jetpack/modules/stats.php]: image

```suggestion
			font-size: 24px;
			line-height: 32px;
```

**Review by jsnmoon — CHANGES_REQUESTED** [review]
These changes look reasonable and test well on my local Jetpack test site. I recommended some styling tweaks and highlighted where we might have accidentally removed two internationalization calls.

I do have a suggestion for simplifying [our background implementation](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Backgrounds_and_Borders/Using_multiple_backgrounds), however:

1. Remove `img.stats-odyssey-notice--image`.
2. Apply `width: 100%;` to both `.stats-odyssey-notice--content` and `.stats-odyssey-notice--image-container`.
3. Replace the existing `background-image` rule with `url("/wp-content/plugins/jetpack/images/odyssey-upgrade/background.png"), url("/wp-content/plugins/jetpack/images/odyssey-upgrade/gradient.png")`.

The result should look like the following:

image

File: projects/plugins/jetpack/modules/stats.php (hunk 1 above)
Comment (a8ck3n) [reply projects/plugins/jetpack/modules/stats.php]: Ah! No reasoning behind this. Just a mistake on my part. I was writing the new layouts side-by-side with the old ones (ie: a new function call) and displaying both in the UI so I could compare the results. I overlooked this when cleaning up and removing the old code path.

Thanks for flagging this!

**Comment by a8ck3n** [comment]
Thanks for the review and suggestions @jsnmoon. Changes incoming!

Regarding the padding and font bits, I was working from the [Notice class](https://github.com/Automattic/jetpack/blob/trunk/projects/js-packages/components/components/notice/style.module.scss) in JS Packages → Components. I should have double-checked against the Figma file once I had things in place though.

**Review by jsnmoon — APPROVED** [review]
Looks great; feel free to ship!

cc @jeffgolenski: The Odyssey Stats screenshot dimensions are 437px x 210px. Should we use a higher-resolution image here, or are we okay with a bit of pixelation?

**Review by jsnmoon — DISMISSED** [review]

**Comment by a8ck3n** [comment]
> Looks great; feel free to ship!
> 
> cc @jeffgolenski: The Odyssey Stats screenshot dimensions are 437px x 210px. Should we use a higher-resolution image here, or are we okay with a bit of pixelation?

@jsnmoon We have higher resolution versions in the original issue.

I downsized them when I thought the max display height would be 210px. We could definitely resize from the original assets again if needed but we probably don't want to use the original assets as is. The gradient image itself was over 3 MB in size. I'll spin up a follow-up issue to review the images though, so we can see if there's a better size/quality trade-off we can hit.

**Comment by a8ck3n** [comment]
**Update:** I restored the original promo image to address the fuzzy look Jason noted. We're currently still using a scaled version of the gradient though, as I'm not seeing a noticeable difference when restoring the original full size image file.

@jeffgolenski — Let me know if this works for you!

**Review by jsnmoon — APPROVED** [review]
Looks good!

**Pull request merged**